src/core/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -0,0 +1,158 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import cx from 'classnames';
+import { useTheme } from '../utils/hooks/useTheme';
+import { CommonProps } from '../utils/props';
+import SvgChevronRight from '@itwin/itwinui-icons-react/cjs/icons/ChevronRight';
+import '@itwin/itwinui-css/css/breadcrumbs.css';
+import { useResizeObserver } from '../utils/hooks/useResizeObserver';
+import { useMergedRefs } from '../utils/hooks/useMergedRefs';
+
+export type BreadcrumbsProps = {
+  /**
+   * Index of the currently active breadcrumb.
+   * Defaults to last index in the items array.
+   */
+  currentIndex?: number;
+  /**
+   * Breadcrumb items.
+   */
+  items: JSX.Element[];
+  /**
+   * Specify a custom separator element to show between breadcrumb items.
+   * Defaults to the `SvgChevronRight` icon.
+   */
+  separator?: React.ReactNode;
+} & Omit<CommonProps, 'title'>;
+
+/**
+ * A breadcrumb trail is used as a navigational aid to help users keep track
+ * of their place in the application. It is often placed before a page's main content.
+ *
+ * For accessibility, make sure to provide an aria-label describing the type of navigation.
+ *
+ * @example
+ * <Breadcrumbs
+ *   items={[
+ *     <BreadcrumbItem onClick={() => {}}>Root</BreadcrumbItem>,
+ *     <BreadcrumbItem onClick={() => {}}>Item 1</BreadcrumbItem>,
+ *     <BreadcrumbItem onClick={() => {}}>Item 2</BreadcrumbItem>,
+ *   ]}
+ *   aria-label='Primary'
+ * />
+ />
+ */
+export const Breadcrumbs = React.forwardRef(
+  (props: BreadcrumbsProps, ref: React.RefObject<HTMLElement>) => {
+    const {
+      items,
+      currentIndex = items.length - 1,
+      separator,
+      className,
+      ...rest
+    } = props;
+
+    useTheme();
+
+    const breadcrumbsRef = React.useRef<HTMLElement>(null);
+
+    const [hiddenItems, setHiddenItems] = React.useState<JSX.Element[]>([]);
```

Comment:
I am just thinking, do you really need `hiddenItems`? You are not showing them anythere. You kind of only need to know how many items to show and can `slice` from original array.

Reply:
Changed to use `visibleCount`